### DISCUSSION TITLE
Concatenate lists in a column?

### DISCUSSION THREAD

**Discussion by seolmor** [cozodb/cozo#147] [Q&A]
Given a column with three rows, each row holding a list, is it possible to concatenate the lists?
```
q1[n] <- [[["apple","pear"]],[["plum"]],[["pear"]]] 
?[collect(y)] := q1[n], y=concat(n)
```
With or without the  `y=concat(n)`  clause, the above outputs -

 [["apple", "pear"], ["pear"], ["plum"]] 

I'm trying to get -

 ["apple", "pear",  "pear", "plum"]

**Comment by matthiasautrata** [comment]
maybe union(...) instead of concat? (set union, not bag, like you asked?)

**Comment by seolmor** [comment]
Thank you.

`union(...)`  produces a set but I want a bag to retain the second occurrence of "pear"
```
q1[n] <- [[["apple","pear"]],[["plum"]],[["pear"]]] 
?[union(n)] := q1[n]
=>
 union(n) 
---------------------------
 ["apple", "pear", "plum"] 
```
